src/meta-srv/src/handler/collect_datanode_cluster_info_handler.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use api::v1::meta::{HeartbeatRequest, Role};
+use common_meta::cluster;
+use common_meta::cluster::{DatanodeStatus, NodeInfo, NodeInfoKey, NodeStatus};
+use common_meta::rpc::store::PutRequest;
+use snafu::ResultExt;
+use store_api::region_engine::RegionRole;
+
+use crate::error::{Result, SaveClusterInfoSnafu};
+use crate::handler::{HandleControl, HeartbeatAccumulator, HeartbeatHandler};
+use crate::metasrv::Context;
+
+pub struct CollectDatanodeClusterInfoHandler;
+
+#[async_trait::async_trait]
+impl HeartbeatHandler for CollectDatanodeClusterInfoHandler {
```

Comment:
The `CollectDatanodeClusterInfoHandler` and `CollectFrontendClusterInfoHandler` logic are almost the same, let's refactor it to be one handler with different configurations.

Reply:
The handler follows the principle of single responsibility.
As DatanodeStatus and FrontendStatus evolve, they will become more different, but I can try to share some methods.